x/mint/depinject.go

```diff
@@ -64,31 +64,22 @@ func ProvideModule(in ModuleInputs) ModuleOutputs {
 		panic(err)
 	}
 
+	if in.MintFn == nil {
+		panic("mintFn cannot be nil")
+	}
+
 	k := keeper.NewKeeper(
 		in.Cdc,
 		in.Environment,
-		in.StakingKeeper,
 		in.AccountKeeper,
 		in.BankKeeper,
 		feeCollectorName,
 		as,
 	)
 
-	if in.MintFn != nil && in.InflationCalculationFn != nil {
```

Comment:
I am not fan of that setter, but we can get an OK UX with depinject.
IMHO we should keep most of this logic and just make the stakingkeeper optional.
If StakingKeeper && MintFn is nil, then we panic as well.
The rest of this can be brought back.